### ISSUE TITLE
CI is failing as the docker service can't be restarted

### ISSUE THREAD

**Issue by fidencio** [kata-containers/tests#3982]
```
10:18:05 #########################################
10:18:05 [setup.sh:203] INFO: restart docker service
10:18:05 [setup.sh:206] WARNING: Service docker is not active
10:18:05 A dependency job for docker.service failed. See 'journalctl -xe' for details.
10:18:05 [setup.sh:228] WARNING: Can't restart the docker service
10:18:05 [jenkins_job_build.sh:170] ERROR: ci/setup.sh
10:18:05 Build step 'Execute shell' marked build as failure
```

Reference: http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-ARM-PR/2337/

@jongwu, can you take a look at this one?

**Comment by fidencio** [comment]
And this is not only related to the ARM CI, as it also happened on the metrics CI:
```
10:36:00 #########################################
10:36:00 [setup.sh:203] INFO: restart docker service
10:36:00 [setup.sh:206] WARNING: Service docker is not active
10:36:00 A dependency job for docker.service failed. See 'journalctl -xe' for details.
10:36:00 [setup.sh:228] WARNING: Can't restart the docker service
10:36:00 [jenkins_job_build.sh:170] ERROR: ci/setup.sh
10:36:00 ++ handle_error 74
10:36:00 ++ local exit_code=1
10:36:00 ++ local line_number=74
10:36:00 ++ echo 'Failed at /home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/ci_entry_point.sh +74: .ci/jenkins_job_build.sh "${repo_to_test}"'
10:36:00 Failed at /home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/ci_entry_point.sh +74: .ci/jenkins_job_build.sh "${repo_to_test}"
10:36:00 ++ exit 1
``` 
Reference: http://jenkins.katacontainers.io/job/kata-containers-2.0-metrics-ubuntu-20-04-PR/1457/

**Comment by fidencio** [comment]
```
10:35:42 Setup env for kata repository: github.com/kata-containers/kata-containers
10:35:42 Cloning into '/home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/go/src/github.com/kata-containers/kata-containers'...
10:35:44 Looking for baremetal cleanup script for arch x86_64
10:35:44 Running common baremetal cleanup script
10:35:45 panic: reflect: indirection through nil pointer to embedded struct [recovered]
10:35:45 	panic: reflect: indirection through nil pointer to embedded struct
10:35:45 
10:35:45 goroutine 1 [running]:
10:35:45 text/template.errRecover(0xc0005b12e8)
10:35:45 	/usr/local/go/src/text/template/exec.go:169 +0x1c5
10:35:45 panic(0x56195ad75020, 0x56195b06e468)
10:35:45 	/usr/local/go/src/runtime/panic.go:965 +0x1b9
10:35:45 reflect.Value.FieldByIndex(0x56195af0c440, 0xc000346580, 0x99, 0xc0001b89a0, 0x2, 0x2, 0x0, 0x56195b0fd2e8, 0x56195ad75020)
10:35:45 	/usr/local/go/src/reflect/value.go:889 +0x325
10:35:45 text/template.(*state).evalField(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0x7fff8fedeb51, 0x6, 0x56195b0e6ea0, 0xc00021e390, 0xc00030e110, 0x1, ...)
10:35:45 	/usr/local/go/src/text/template/exec.go:617 +0x4ed
10:35:45 text/template.(*state).evalFieldChain(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0x56195af0c440, 0xc000346580, 0x99, 0x56195b0e6ea0, 0xc00021e390, 0xc00030e100, ...)
10:35:45 	/usr/local/go/src/text/template/exec.go:570 +0x225
10:35:45 text/template.(*state).evalFieldNode(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0xc00021e390, 0xc00030e110, 0x1, 0x1, 0x56195ae030c0, 0x56195be819d0, ...)
10:35:45 	/usr/local/go/src/text/template/exec.go:534 +0x115
10:35:45 text/template.(*state).evalCommand(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0xc00021e360, 0x56195ae030c0, 0x56195be819d0, 0x99, 0xd0, 0xd0, ...)
10:35:45 	/usr/local/go/src/text/template/exec.go:462 +0x8f0
10:35:45 text/template.(*state).evalPipeline(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0xc0000aee40, 0xc000222228, 0x0, 0xc0005b1160)
10:35:45 	/usr/local/go/src/text/template/exec.go:436 +0x125
10:35:45 text/template.(*state).walk(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0x56195b0e6ca8, 0xc00021e3c0)
10:35:45 	/usr/local/go/src/text/template/exec.go:255 +0x38a
10:35:45 text/template.(*state).walk(0xc0005b1258, 0x56195af0c440, 0xc000346580, 0x99, 0x56195b0e6f78, 0xc00021e330)
10:35:45 	/usr/local/go/src/text/template/exec.go:264 +0x13c
10:35:45 text/template.(*Template).execute(0xc000346500, 0x56195b08d998, 0xc0003097a0, 0x56195af0c440, 0xc000346580, 0x0, 0x0)
10:35:45 	/usr/local/go/src/text/template/exec.go:220 +0x1c6
10:35:45 text/template.(*Template).Execute(...)
10:35:45 	/usr/local/go/src/text/template/exec.go:203
10:35:45 github.com/docker/cli/cli/command/system.formatInfo(0x56195b0fa0e8, 0xc00050d6c0, 0x0, 0xc0004d9b70, 0x1, 0x1, 0xc00040e550, 0x0, 0x0, 0x0, ...)
10:35:45 	/go/src/github.com/docker/cli/cli/command/system/info.go:487 +0x1fd
10:35:45 github.com/docker/cli/cli/command/system.runInfo(0xc00077b080, 0x56195b0fa0e8, 0xc00050d6c0, 0xc0004d9420, 0x0, 0x0)
10:35:45 	/go/src/github.com/docker/cli/cli/command/system/info.go:87 +0x3a5
10:35:45 github.com/docker/cli/cli/command/system.NewInfoCommand.func1(0xc00077b080, 0xc0004d9940, 0x0, 0x1, 0x0, 0x0)
10:35:45 	/go/src/github.com/docker/cli/cli/command/system/info.go:53 +0x48
10:35:45 github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc00077b080, 0xc00052f0d0, 0x1, 0x1, 0xc00077b080, 0xc00052f0d0)
10:35:45 	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:850 +0x472
10:35:45 github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc000164b00, 0xc00052f0c0, 0x2, 0x2)
10:35:45 	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:958 +0x375
10:35:45 github.com/docker/cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
10:35:45 	/go/src/github.com/docker/cli/vendor/github.com/spf13/cobra/command.go:895
10:35:45 main.runDocker(0xc00050d6c0, 0x56195b08ff38, 0xc000134010)
10:35:45 	/go/src/github.com/docker/cli/cmd/docker/docker.go:287 +0x1eb
10:35:45 main.main()
10:35:45 	/go/src/github.com/docker/cli/cmd/docker/docker.go:298 +0xfc
10:35:45 [clean_up.sh:335] INFO: kill stale process
10:35:45 Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
10:35:45 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:45 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:45 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:45 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:45 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:45 /home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/go/src/github.com/kata-containers/tests/lib/common.bash: line 91: --version: command not found
10:35:46 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:46 /etc/kata-containers/configuration.toml: Either initrd or image must be set to a valid path (initrd: initrd is not set) (image: file /usr/share/kata-containers/kata-containers.img does not exist)
10:35:46 [clean_up.sh:338] INFO: delete stale docker resource under /var/lib/docker
10:35:47 Synchronizing state of docker.service with SysV service script with /lib/systemd/systemd-sysv-install.
10:35:47 Executing: /lib/systemd/systemd-sysv-install disable docker
10:35:48 [clean_up.sh:341] INFO: remove containers started by ctr
10:35:48 [clean_up.sh:343] INFO: delete stale kata resource under /var/lib/vc /run/vc /usr/share/kata-containers /usr/share/defaults/kata-containers
10:35:48 [clean_up.sh:345] INFO: Remove installed kata packages
10:35:48 ERROR: packages not installed
10:35:48 [clean_up.sh:347] INFO: Remove installed cri-o related binaries and configuration
10:35:48 [clean_up.sh:349] INFO: Remove installed containerd-cri related binaries and configuration
10:35:48 rm: cannot remove '/var/lib/containerd-test': No such file or directory
10:35:48 rm: cannot remove '/run/containerd-test': No such file or directory
10:35:48 [clean_up.sh:353] INFO: Remove installed kubernetes packages and configuration
10:35:48 E: Could not get lock /var/lib/dpkg/lock-frontend. It is held by process 288847 (apt-get)
10:35:48 E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), is another process using it?
10:35:48 [clean_up.sh:365] INFO: Clean up stale network interface
10:35:48 [clean_up.sh:368] INFO: Remove Kata package repo registrations
10:35:48 Warning: apt-key output should not be parsed (stdout is not a terminal)
10:35:48 [clean_up.sh:264] INFO: No katacontainers key found - not removing
10:35:48 [clean_up.sh:371] INFO: Clean GOCACHE
10:35:48 [clean_up.sh:380] INFO: Clean transient test data and logs which has been stored under /var/log/kata-tests
10:35:48 ~/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/go/src/github.com/kata-containers/kata-containers ~/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/go/src/github.com/kata-containers/tests
10:35:48 /tmp/install-go-tmp.glQpierSLo ~/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/go/src/github.com/kata-containers/kata-containers
```

**Comment by GabyCT** [comment]
@fidencio the problem is related with the agent-tracing test 
```
05:01:35 
05:01:35 [test-agent-shutdown.sh:595] INFO: agent log file
05:01:35 
05:01:35 [test-agent-shutdown.sh:604] INFO: OCI bundle directory : /tmp/tmp.tMPCSSEG5h
05:01:35 [jenkins_job_build.sh:196] ERROR: ci/run.sh
```
@cmaf @jodh-intel

**Comment by Jakob-Naucke** [comment]
@GabyCT not always http://jenkins.katacontainers.io/job/kata-containers-2.0-metrics-ubuntu-20-04-PR/1508:

```
GOPATH=/home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/go
CODECOV_TOKEN=****
_=/usr/bin/env
#########################################
[setup.sh:203] INFO: restart docker service
[setup.sh:206] WARNING: Service docker is not active
A dependency job for docker.service failed. See 'journalctl -xe' for details.
[setup.sh:228] WARNING: Can't restart the docker service
[jenkins_job_build.sh:170] ERROR: ci/setup.sh
++ handle_error 74
++ local exit_code=1
++ local line_number=74
++ echo 'Failed at /home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/ci_entry_point.sh +74: .ci/jenkins_job_build.sh "${repo_to_test}"'
Failed at /home/jenkins/workspace/workspace/kata-containers-2.0-metrics-ubuntu-20-04-PR/ci_entry_point.sh +74: .ci/jenkins_job_build.sh "${repo_to_test}"
++ exit 1
```

**Comment by cmaf** [comment]
@jongwu, maybe @jcvenegas, could you take a look? The failure occurs on the ARM CI, metrics CI

**Comment by jongwu** [comment]
Hello @cmaf -, I see that Arm CI is stable these days. can you give link of the failure of Arm CI?

**Comment by GabyCT** [comment]
The issue on metrics has been solved by https://github.com/kata-containers/tests/commit/a909d7530544ad4f5bfa81e6e1595321c741ae17

**Comment by Jakob-Naucke** [comment]
> The issue on metrics has been solved by a909d7530544ad4f5bfa81e6e1595321c741ae17

@GabyCT I'm afraid it hasn't, PTAL @ http://jenkins.katacontainers.io/job/kata-containers-2.0-metrics-ubuntu-20-04-PR/1582
